subgraphs/dopex/src/entities/pool.ts

```diff
@@ -0,0 +1,558 @@
+  amountChange: BigInt,
+  eventType: EventType
+): void {
+  const collateralToken = getOrCreateToken(
+    event,
+    Address.fromBytes(pool.inputTokens[0])
+  );
+  const inputTokenBalances = pool.inputTokenBalances;
+  switch (eventType) {
+    case EventType.Deposit:
+    case EventType.Purchase:
+      inputTokenBalances[0] = inputTokenBalances[0].plus(amountChange);
+      break;
+    case EventType.Withdraw:
+    case EventType.Settle:
+      inputTokenBalances[0] = inputTokenBalances[0].minus(amountChange);
+      break;
+    default:
+      break;
+  }
+  pool.inputTokenBalances = inputTokenBalances;
+
+  const prevPoolTVL = pool.totalValueLockedUSD;
+  pool.totalValueLockedUSD = convertTokenToDecimal(
+    pool.inputTokenBalances[0],
+    collateralToken.decimals
+  ).times(collateralToken.lastPriceUSD!);
+  const tvlChangeUSD = pool.totalValueLockedUSD.minus(prevPoolTVL);
+
+  pool._lastUpdateTimestamp = event.block.timestamp;
+  pool.save();
+
+  // Protocol
+  updateProtocolTVL(event, tvlChangeUSD);
+}
+
+export function updatePoolOpenInterestUSD(
+  event: ethereum.Event,
+  pool: LiquidityPool,
+  amountChangeUSD: BigDecimal,
+  isIncrease: boolean
+): void {
+  if (isIncrease) {
+    pool.openInterestUSD = pool.openInterestUSD.plus(amountChangeUSD);
+    pool._cumulativeOpenInterestUSD =
+      pool._cumulativeOpenInterestUSD.plus(amountChangeUSD);
+  } else {
+    pool.openInterestUSD = pool.openInterestUSD.minus(amountChangeUSD);
+  }
+
+  pool._lastUpdateTimestamp = event.block.timestamp;
+  pool.save();
+
+  // Protocol
+  updateProtocolOpenInterestUSD(event, amountChangeUSD, isIncrease);
+}
+
+export function UpdatePoolOpenPositionCount(
+  event: ethereum.Event,
+  pool: LiquidityPool,
+  isIncrease: boolean
+): void {
+  if (isIncrease) {
+    pool.openPositionCount += INT_ONE;
+  } else {
+    pool.openPositionCount -= INT_ONE;
+    pool.closedPositionCount += INT_ONE;
+  }
+
+  pool._lastUpdateTimestamp = event.block.timestamp;
+  pool.save();
+
+  updateProtocolOpenPositionCount(event, isIncrease);
+}
+
+export function incrementPoolEventCount(
+  event: ethereum.Event,
+  pool: LiquidityPool,
+  eventType: EventType
+): void {
+  const isPut = pool._isPut;
+  switch (eventType) {
+    case EventType.Deposit:
+      if (isPut) {
+        pool.putsMintedCount += INT_ONE;
+      } else {
+        pool.callsMintedCount += INT_ONE;
+      }
+      pool.contractsMintedCount += INT_ONE;
+      break;
+    case EventType.Withdraw:
+      pool.contractsExpiredCount += INT_ONE;
+      pool.contractsClosedCount += INT_ONE;
+      break;
+    case EventType.Purchase:
+      pool.contractsTakenCount += INT_ONE;
+      break;
+    case EventType.Settle:
+      pool.contractsExercisedCount += INT_ONE;
+      break;
+    default:
+      break;
+  }
+  pool._lastUpdateTimestamp = event.block.timestamp;
+  pool.save();
+
+  incrementProtocolEventCount(event, eventType, isPut);
+}
+
+export function updatePoolInputTokenBalance(
```

Comment:
Consider renaming the function to `addPoolInputTokenBalance`, since `update` might be confused with an overwrite of the current value with the one passed to the function

Reply:
Change to name to increasePoolInputTokenBalance (use increase to be consistent with isIncrease flag in the parameters) at 549f81dbabc769cba506e8859c380881f1c9d64c